cvat/apps/dataset_manager/views.py

```diff
@@ -133,41 +154,52 @@ def export(dst_format, project_id=None, task_id=None, job_id=None, server_url=No
 
         os.makedirs(cache_dir, exist_ok=True)
 
+        # acquire a lock 2 times instead of using one long lock:
+        # 1. to check whether the file exists or not
+        # 2. to create a file when it doesn't exist
         with get_export_cache_lock(
             output_path,
-            block=True,
-            acquire_timeout=EXPORT_CACHE_LOCK_TIMEOUT,
-            ttl=rq.get_current_job().timeout,
+            ttl=EXPORT_LOCK_TTL,
+            acquire_timeout=EXPORT_CACHE_LOCK_ACQUISITION_TIMEOUT,
         ):
-            if not osp.exists(output_path):
-                with tempfile.TemporaryDirectory(dir=cache_dir) as temp_dir:
-                    temp_file = osp.join(temp_dir, 'result')
-                    export_fn(db_instance.id, temp_file, dst_format,
-                        server_url=server_url, save_images=save_images)
-                    os.replace(temp_file, output_path)
-
-                scheduler: Scheduler = django_rq.get_scheduler(
-                    settings.CVAT_QUEUES.EXPORT_DATA.value
-                )
-                cleaning_job = scheduler.enqueue_in(
-                    time_delta=cache_ttl,
-                    func=clear_export_cache,
-                    file_path=output_path,
-                    file_ctime=instance_update_time.timestamp(),
-                    logger=logger
-                )
-                logger.info(
-                    "The {} '{}' is exported as '{}' at '{}' "
-                    "and available for downloading for the next {}. "
-                    "Export cache cleaning job is enqueued, id '{}'".format(
-                        db_instance.__class__.__name__.lower(),
-                        db_instance.name if isinstance(
-                            db_instance, (Project, Task)
-                        ) else db_instance.id,
-                        dst_format, output_path, cache_ttl,
-                        cleaning_job.id
-                    )
-                )
+            if osp_exists(output_path):
+                # Update last modification time to prolong the export lifetime
+                # and postpone the file deleting by the cleaning job
+                os.utime(output_path, None)
```

Comment:
Now as we call it from several places, it makes sense to factor out a function for this to encapsulate the logic. https://github.com/cvat-ai/cvat/blob/2bf9a0bae0e39fabdee9e9f4ca82b63cb3be7174/cvat/apps/engine/background.py#L334

Reply:
done